pkg/controller/ibmblockcsi/ibmblockcsi_controller.go

```diff
@@ -535,46 +534,38 @@ func (r *ReconcileIBMBlockCSI) getRestartedAtAnnotation(Annotations map[string]s
 	return "", ""
 }
 
-func (r *ReconcileIBMBlockCSI) getControllerK8sObject(instance *ibmblockcsi.IBMBlockCSI) (error, *appsv1.StatefulSet) {
+func (r *ReconcileIBMBlockCSI) getControllerK8sObject(instance *ibmblockcsi.IBMBlockCSI) (*appsv1.StatefulSet, error) {
 			controllerStatefulset := &appsv1.StatefulSet{}
 			err := r.client.Get(context.TODO(), types.NamespacedName{
 				Name:      oconfig.GetNameForResource(oconfig.CSIController, instance.Name),
 				Namespace: instance.Namespace,
 			}, controllerStatefulset)
-		
-			if err != nil {
-				return err, controllerStatefulset
-			}
-			return err, controllerStatefulset
+
+			return controllerStatefulset, err
 }
 
-func (r *ReconcileIBMBlockCSI) getNodeK8sObject(instance *ibmblockcsi.IBMBlockCSI) (error, *appsv1.DaemonSet) {
+func (r *ReconcileIBMBlockCSI) getNodeK8sObject(instance *ibmblockcsi.IBMBlockCSI) (*appsv1.DaemonSet, error) {
 	node := &appsv1.DaemonSet{}
 	err := r.client.Get(context.TODO(), types.NamespacedName{
 		Name:      oconfig.GetNameForResource(oconfig.CSINode, instance.Name),
 		Namespace: instance.Namespace,
 	}, node)
 
-	if err != nil {
-		return err, node
-	}
-	return err, node
+	return node, err
 }
 
-func (r *ReconcileIBMBlockCSI) getDriverPodsStatus(instance *ibmblockcsi.IBMBlockCSI, 
-	controller *appsv1.StatefulSet, node *appsv1.DaemonSet) (bool, bool) {
+func (r *ReconcileIBMBlockCSI) getControllerPodsStatus(instance *ibmblockcsi.IBMBlockCSI, controller *appsv1.StatefulSet) bool {
 	ControllerReady := false
+
+	ControllerReady = controller.Status.ReadyReplicas == controller.Status.Replicas
+	return ControllerReady
```

Comment:
Why do we need the local variable? 
```suggestion
	return controller.Status.ReadyReplicas == controller.Status.Replicas 
```

Reply:
Also, the function name says that it returns the status, but actually, we return something else.
Rename (also for the node function)